WordPress/gutenberg
# Mobile - Block Actions Menu - Fix block title regression and adds integration tests

Fixes https://github.com/WordPress/gutenberg/issues/46673
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/5337

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5340

## What?
After some changes were introduced in https://github.com/WordPress/gutenberg/pull/43849 related to the refactor of the `lodash` library effort, the block title within the block actions menu picker was showing as `undefined`.

## Why?
To fix a regression in the mobile editor related to the block actions menu.

## How?
By simplifying the code by passing just the current `cliendId` as a string instead of an array with just one `clientId`, by updating this and adapting the code, it now gets the right block title in the block actions menu.

It also adds several integration tests:

  **Block Actions Menu**
    ✓ renders the block name in the Picker's header
    **moving blocks** (from the block actions menu, these show up when the canvas is smaller so we hide the up/down arrows below blocks and show them in the actions menu instead)
      ✓ moves blocks up and down
      ✓ disables the Move Up button for the first block
      ✓ disables the Move Down button for the last block
    **block options**
      ✓ copies and pastes a block
      ✓ does not replace a non empty Paragraph block when pasting another block
      ✓ cuts and pastes a block
      ✓ duplicates a block
      ✓ transforms a Paragraph block into a Pullquote block

  **Block Mover Picker**
    **moving blocks** (The up/down arrows that show up below a block)
      ✓ moves blocks up and down
      ✓ disables the Move Up button for the first block
      ✓ disables the Move Down button for the last block

## Testing Instructions

Even though there are new integration tests that should cover quite a few cases, let's smoke-test using the demo app as well.

- Open the app
- Add a block
- Tap on the `...` three dots button to open the block actions menu
- **Expect** to see the block title's in the header of the picker
- Add several blocks and test: Copy, Cut, Paste after, Duplicate and transform features.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 
Before | After
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/208882495-81d8770e-565c-4045-a5de-42cc96e51dea.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/208880378-1ce6531e-7584-4539-9a32-c08ea6d1078e.png" width="200" /></kbd>
